scripts/chgres_cube.py

```diff
@@ -0,0 +1,204 @@
+        required=True,
+        type=lambda s: s.split("."),
+    )
+    parser.add_argument(
+        "--member",
+        default="000",
+        help="The 3-digit ensemble member number.",
+    )
+    return parser.parse_args(argv)
+
+
+# pylint: disable-next=too-many-locals, too-many-statements
+def run_chgres_cube(config_file, cycle, key_path, member):
+    """
+    Setup and run the chgres_cube Driver.
+    """
+    expt_config = get_yaml_config(config_file)
+
+    # The experiment config will have {{ CRES | env }} expressions in it that need to be
+    # dereferenced during driver initialization
+    cres = expt_config["workflow"]["CRES"]
+    os.environ["CRES"] = cres
+    os.environ["MEMBER"] = member
+
+    # Find the names of the external model files
+    expt_config.dereference(
+        context={
+            "cycle": cycle,
+            **os.environ,
+            **expt_config,
+        }
+    )
+    chgres_cube_config = _walk_key_path(expt_config, key_path)
+    input_type = chgres_cube_config["chgres_cube"]["namelist"]["update_values"][
+        "config"
+    ].get("input_type")
+
+    varsfilepath = chgres_cube_config["input_files_metadata_path"]
+    external_config = get_yaml_config(varsfilepath)
+    external_config_fns = external_config["external_model_fns"]
+    external_config_fhrs = external_config["external_model_fhrs"]
+
+    if "task_make_ics" in key_path:
+        if input_type == "grib2":
+            os.environ["fn_grib2"] = external_config_fns[0]
+        else:
+            os.environ["fn_atm"] = external_config_fns[0]
+            os.environ["fn_sfc"] = external_config_fns[1]
+
+        chgres_cube_driver = ChgresCube(
+            config=config_file,
+            cycle=cycle,
+            key_path=key_path,
+            leadtime=dt.timedelta(hours=0),
+        )
+        rundir = Path(chgres_cube_driver.config["rundir"])
+        logging.info(f"Will run in {rundir}")
+        chgres_cube_driver.run()
+
+        if not (rundir / "runscript.chgres_cube.done").is_file():
+            logging.error("Error occurred running chgres_cube. See component error logs.")
+            sys.exit(1)
+
+        # Deliver output data to the forecast's INPUT dir
+        delivery_dir = rundir.parent / "INPUT"
+        _deliver_files(config=expt_config, dst_dir=delivery_dir, key_path=key_path, src_dir=rundir)
+
+    else: # loop over make_lbcs tasks
+        # This loop will need a version of the config that is not dereferenced
+        expt_config = get_yaml_config(config_file)
+
+        fhrs_and_fns = list(zip(external_config_fhrs, external_config_fns))
+        for external_fhr, external_fn in fhrs_and_fns:
+            if input_type == "grib2":
+                os.environ["fn_grib2"] = external_fn
+            else:
+                os.environ["fn_atm"] = external_fn
+
+            lbc_offset_fhrs = expt_config["task_get_extrn_lbcs"]["envvars"][
+                    "EXTRN_MDL_LBCS_OFFSET_HRS"
+                ]
```

Comment:
```suggestion
            lbc_offset_fhrs = _walk_key_path(expt_config, ["task_get_extrn_lbcs", "envvars", "EXTRN_MDL_LBCS_OFFSET_HRS"])
```
Does this work? It seems a little easier on the eyes to me. If this and my other suggestions to use `_walk_key_path()` more work and are accepted, maybe that function could even be given a shorter name, like `_select()` or even just `_get()`.

Reply:
It should work, makes sense to me and removing hanging brackets is always good